### PR TITLE
Use u64 behind the scenes for solana-tokens

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Floating-point math inaccuracies can cause issues, particularly when working with 9-decimal numbers (as in the spl-feature-proposal use case).

#### Summary of Changes
- Store allocation amounts as u64
- Expect spl-token CSV to provide raw amounts (no decimals)
- Display amounts are still converted to floating point numbers, so there can be inaccuracies in the display. But all critical math (eg. transfer amounts, balance checks) uses integers.

Update of spl-feature-proposal to follow

### CODE REVIEW

File: tokens/src/commands.rs [hunk 1]
```diff
@@ -817,16 +852,25 @@ pub fn test_process_distribute_stake_with_client(client: &RpcClient, sender_keyp
...
         .unwrap();
 
+    let expected_amount = sol_to_lamports(1000.0);
     let alice_pubkey = solana_sdk::pubkey::new_rand();
-    let allocation = Allocation {
-        recipient: alice_pubkey.to_string(),
-        amount: 1000.0,
-        lockup_date: "".to_string(),
-    };
+    // let allocation = Allocation {
```
Comment (mvines) [inline tokens/src/commands.rs:857]: Is this intention?

File: tokens/src/commands.rs [hunk 2]
```diff
@@ -817,16 +852,25 @@ pub fn test_process_distribute_stake_with_client(client: &RpcClient, sender_keyp
...
-    wtr.serialize(&allocation).unwrap();
+    wtr.write_record(&["recipient", "amount", "lockup_date"])
+        .unwrap();
+    wtr.write_record(&[
+        alice_pubkey.to_string(),
+        lamports_to_sol(expected_amount).to_string(),
+        "".to_string(),
+    ])
+    .unwrap();
+    // wtr.serialize(&allocation).unwrap();
```
Comment (mvines) [inline tokens/src/commands.rs:873]: Is this intention?

**Review by mvines — APPROVED** [review]
lgtm overall!  Just those comments that maybe should be removed?

File: tokens/src/commands.rs (hunk 1 above)
Comment (CriesofCarrots) [reply tokens/src/commands.rs:857]: Nope! Sec...

**Pull request merged**